src/EditorFeatures/CSharpTest/GenerateVariable/GenerateVariableTests.cs

```diff
@@ -9336,5 +9336,12 @@ internal MyException(int error, int offset, string message) : base(message)
     string.Format(FeaturesResources.Generate_parameter_0, "Error", "MyException"),
 });
         }
+
+        [WorkItem(48172, "https://github.com/dotnet/roslyn/issues/48172")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsGenerateVariable)]
+        public async Task TestMissingOfferParameterInTopLevel()
+        {
+            await TestMissingInRegularAndScriptAsync("[|Console|].WriteLine();");
```

Comment:
@CyrusNajmabadi The test is passing for regular, but failing for script.

The current behavior for script is not offering to generate parameter but offering to generate fields and property.

I assume the "script" means those files with extension csx. I haven't tried them before, so I don't know what the correct behavior is.